src/main/java/com/github/manolo8/darkbot/backpage/BackpageManager.java

```diff
@@ -256,6 +258,15 @@ private String sidStat() {
         }
     }
 
+    @Override
+    public boolean isInstanceValid() {
+        // in isInvalid() method, checking sid & instance from stats manager are not synchronised
+        // so even if #isInvalid() returns false the connection can throw exception anyway
+        // instead local sid & instance will be updated once before any task tick
+        // or even better, keep old vales if new are empty, where old ones can still be valid
```

Comment:
```suggestion
        // Only check against local sid & instance variables, since stats manager ones are
        // updated in the main thread, while the variables here are updated on the background
        // thread every tick
```
I'd simplify this comment, it was phrased a bit weird